### DISCUSSION TITLE
[Feature Request] Text stroke

### DISCUSSION THREAD

**Discussion by bergstar** [tailwindlabs/tailwindcss#1056] [Ideas]
Looks like its pretty well supported:
https://caniuse.com/#search=text-stroke
... and there is a shadow fallback method.

Can sometimes look tacky, but a very useful feature when used properly:
https://www.kirupa.com/tricks/text_outline_effect.htm

**Comment by hacknug** [comment]
I created a plugin a while ago for this. If you have any suggestions, please, let me know! https://github.com/hacknug/tailwindcss-text-fill-stroke

**Comment by jesse-deboer** [comment]
This is now supported on all major browser with a `-webkit` prefix!

It would be a nice feature to add to Tailwind, maybe as a `text-stroke` where you can variate with `text-stroke-2` for example and colors with `text-stroke-white`?

**Reply by hacknug** [reply]
Literally what my plugin does 😄

**Reply by jesse-deboer** [reply]
Yes, I know, but it should also be in the Tailwind core I think! 😄

**Reply by faizananwerali** [reply]
> Literally what my plugin does 😄

Lots of core feature is lacking in your plugin, that's why I removed after adding and tested in one of my project.

like:
1. custom color for text stroke.
2. arbitrary width for text stroke.

Just look at these regular stroke tailwind utility classes used for SVG. [https://tailwindcss.com/docs/stroke-width](https://tailwindcss.com/docs/stroke-width) [https://tailwindcss.com/docs/stroke-width](https://tailwindcss.com/docs/stroke-width)

You will get an idea of what's missing.

**Reply by hacknug** [reply]
@faizananwerali have you tried v2? I added support for arbitrary values in fill and stroke properties. Values have always been customizable via your config file. If something is not working as expected, please let me know by opening an issue in my repo 👍

**Reply by faizananwerali** [reply]
> @faizananwerali have you tried v2? I added support for arbitrary values in fill and stroke properties. Values have always been customizable via your config file. If something is not working as expected, please let me know by opening an issue in my repo 👍

What version 2? I don't see any v2 in [https://github.com/hacknug/tailwindcss-text-fill-stroke](https://github.com/hacknug/tailwindcss-text-fill-stroke) on npm page. The last version is 1.1.2 from three years ago. It would be great if you could share any link if possible.

**Reply by hacknug** [reply]
@faizananwerali Here you go: https://www.npmjs.com/package/tailwindcss-text-fill-stroke/v/2.0.0-beta.1

**Comment by dcheglakov** [comment]
In Tailwind 4 I've created custom utility 
```css
@utility text-stroke-* {
   -webkit-text-stroke-width: calc(--value(integer) * 1px);
   -webkit-text-stroke-color: --value(--color-*);
   -webkit-text-stroke-color: --value([*]);
}
```
**Example**
`text-stroke-gray-100 text-stroke-1`
`text-stroke-[#454545] text-stroke-2`
